src/context.h

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright (c) 2019, Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "data.h"
+#include <map>
+#include <string>
+
+namespace yarpgen {
+
+// Class that is used to determine the evaluation context.
+// It allows us to evaluate the same arithmetic tree with different input values.
+class EvalCtx {
```

Comment:
Not sure that I understand the idea here, but ok as it's just a placeholder for now.

Reply:
The idea here is to be able to evaluate the same expression with different values without actually changing them. It allows us to override them within one arithmetic expression.